tests-beta/android/MASVS-RESILIENCE/MASTG-TEST-0x39-2.md

```diff
@@ -0,0 +1,31 @@
+---
+title: Debugging enabled for WebViews
+platform: android
+id: MASTG-TEST-0x39-2
+type: [static]
+weakness: MASWE-0067
+---
+
+## Overview
+
+To enable debugging of Webviews, the API `WebView.setWebContentsDebuggingEnabled(true);` can be used to enable WebView debugging (see ["see Android documentation"](https://developer.chrome.com/docs/devtools/remote-debugging/webviews/#configure_webviews_for_debugging)).
+
+Calling this API will enable WebView debugging for **all** of the application's WebViews, independently of the AndroidManifest.xml debuggable flag (See @MASTG-TEST-0x39-1). This allows an attached PC to eavesdrop on, and modify all communication inside WebViews.
+
+!!! note Other ways to debug the application
+    Disabling WebView debugging does not fully prevent all possibilities to debug the app and the WebViews. See @MASWE-0101 for more details on how to prevent debugging.
+
+## Steps
+
+1. Reverse engineer the app (@MASTG-TECH-0017).
+2. Inspect all locations where `WebView.setWebContentsDebuggingEnabled` is called, and check
+    1. if it is set to true, and
+    2. if the API call can be executed during runtime.
```

Comment:
```suggestion
2. Obtain all locations where `WebView.setWebContentsDebuggingEnabled` is called and set to `true`.
```

Reply:
what's meant by "if the API call can be executed during runtime"?

Reply:
https://developer.chrome.com/docs/devtools/remote-debugging/webviews/#configure_webviews_for_debugging contains a code snippet that ties the API call to the manifest debuggable flag.

What this sentence should mean is that the API call itself is not a problem if it is essentially in dead code. Maybe it still should not be in the app...
```
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
    if (0 != (getApplicationInfo().flags & ApplicationInfo.FLAG_DEBUGGABLE))
    { WebView.setWebContentsDebuggingEnabled(true); }
}
```

Reply:
Ok, got it! Let's see if we should reformulate and/or use this for the evaluation and remediation as well but with more detail mentioning those conditions.